packages/core/src/components/forms/controls.tsx

```diff
@@ -31,22 +34,36 @@ export interface IControlProps extends IProps {
     /** Ref handler that receives HTML `<input>` element backing this component. */
     inputRef?: (ref: HTMLInputElement) => any;
 
-    /** Text label for the control. */
+    /**
+     * Text label for the control.
+     *
+     * This prop actually supports JSX elements, but TypeScript will throw an error because
+     * `HTMLProps` only allows strings. Use `labelElement` to supply a JSX element in TypeScript.
+     */
     label?: string;
 
+    /**
+     * JSX Element label for the control.
+     *
+     * This prop is necessary for TypeScript consumers as the type definition for `label` only
+     * accepts strings. JavaScript consumers can provide a JSX element directly to `label`.
+     */
+    labelElement?: React.ReactNode;
+
     /** Event handler invoked when input value is changed. */
     onChange?: React.FormEventHandler<HTMLInputElement>;
 }
 
-const INVALID_PROPS = [
+const INVALID_PROPS: Array<keyof ICheckboxProps> = [
     // we spread props to `<input>` but render `children` as its sibling
     "children",
     "defaultIndeterminate",
     "indeterminate",
+    "labelElement",
 ];
 
 /** Base Component class for all Controls */
-export class Control<P extends IControlProps> extends React.Component<React.HTMLProps<HTMLInputElement> & P, {}> {
+export class Control<P extends IControlProps> extends React.Component<P, {}> {
```

Comment:
Do we still need `<P extends IControlProps>` if we already have interfaces extending `IControlProps`, such as `export interface ICheckboxProps extends IControlProps`?